filter/ArticleMedraXmlFilter.inc.php

```diff
@@ -331,6 +405,21 @@ function createContentItemNode($doc, $issue, $article, $galley, $objectLocalePre
 			}
 
 		}
+		//Adding Citation List (unstructured) on the content
+		$citationDao = DAORegistry::getDAO('CitationDAO'); /* @var $citationDao CitationDAO */
+		$parsedCitations = $citationDao->getBySubmissionId($article->getId())->toArray();
```

Comment:
There is no such function, but getByPublicationId(), s. https://github.com/pkp/pkp-lib/blob/stable-3_3_0/classes/citation/CitationDAO.inc.php#L111

Reply:
Because I used `$parsedCitations = $citationDao->getBySubmissionId($article->getId())->toArray();` in a former release of medra plugin, I tried the following function:
`/**
	 * Retrieve an array of citations matching a particular publication id.
	 * @param $publicationId int
	 * @param $dbResultRange DBResultRange the desired range
	 * @return DAOResultFactory containing matching Citations
	 */
	function getByPublicationId($publicationId, $rangeInfo = null) {
		$result = $this->retrieveRange(
			'SELECT *
			FROM citations
			WHERE publication_id = ?
			ORDER BY seq, citation_id',
			[(int)$publicationId],
			$rangeInfo
		);
		return new DAOResultFactory($result, $this, '_fromRow', ['id']);
	}`
but in the old version that function is not available.

Reply:
I think this would be: `$citationDao->getByPublicationId($article->getCurrentPublication()->getId())`